Vegeting/Global/UIComponent/InterestView.swift

```diff
@@ -0,0 +1,134 @@
+    private let entryPoint: EntryPoint
+    
+    weak var delegate: InterestViewDelegate?
+    
+    // MARK: - init
+    
+    init(interestList: [String], entryPoint: EntryPoint) {
+        self.interestList = interestList
+        self.entryPoint = entryPoint
+        super.init(frame: .zero)
+        configureCollectionView()
+        setupLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - func
+    
+    private func setupLayout() {
+        addSubview(interestCollectionView)
+        interestCollectionView.constraint(to: self)
+    }
+    
+    private func configureCollectionView() {
+        interestCollectionView.dataSource = self
+        interestCollectionView.delegate = self
+    }
+    
+    func changeCategoryList(with list: [String]) {
+        interestList = list
+    }
+}
+
+extension InterestView: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return interestList.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: InterestCollectionViewCell.className, for: indexPath) as? InterestCollectionViewCell else { return UICollectionViewCell() }
+        
+        cell.configure(with: interestList[indexPath.item])
+        return cell
+    }
+}
+
+extension InterestView: UICollectionViewDelegate {
+    
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        judgeSelectCellEnabled(with: 0)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, shouldDeselectItemAt indexPath: IndexPath) -> Bool {
+        judgeSelectCellEnabled(with: 1)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        judgeBottomButtonEnabled(to: true, with: 1)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        judgeBottomButtonEnabled(to: false, with: 0)
+    }
+    
+    private func judgeBottomButtonEnabled(to isEnabled: Bool, with comparisonValue: Int) {
+        guard let selectedCellCount = interestCollectionView.indexPathsForSelectedItems?.count else { return }
+        if selectedCellCount == comparisonValue {
+            delegate?.setBottomButtonEnabled(to: isEnabled)
+        }
+    }
+    
+    private func judgeSelectCellEnabled(with comparisonValue: Int) -> Bool {
+        guard let selectedCellCount = interestCollectionView.indexPathsForSelectedItems?.count else { return true }
+        
+        if comparisonValue <= selectedCellCount  && selectedCellCount <= comparisonValue + 2 {
+            return true
+        } else {
+            return false
+        }
+    }
```

Comment:
p2;
comparisonValue가 무엇을 의미하는지 알기 어려울 거 같아요. (to등으로 하는건 가독성을 저하시킬 수 있는 거 같습니다!)
p4;
무엇과 비교한다는 걸까?라는 의문이 드는거 같아요.    
cell에 주는 action에 따라서 다른 비교방식을 쓰는걸 enum으로 바꿨을때 가독성이 더 좋은 것 같은데 어떠신가요?    
개인적인 기호인거 같아서, 그대로 두는게 나은거 같으면 반영하지 않으셔도 될 거 같아요.
```suggestion
   enum SelectType {
     case selectCell
     case deselectCell
     var buttonEnableQualify: Int {
       switch self {
       case .selectCell:
         return 0
       case .deselectCell:
         return 1
       
       }
     }
   }
    private func judgeSelectCellEnabled(type: SelectType) -> Bool {
        guard let selectedCellCount = interestCollectionView.indexPathsForSelectedItems?.count else { return true }
        if type.minimumQualify  <= selectedCellCount  && selectedCellCount <= type.minimumQualify  + 2 {
            return true
        } else {
            return false
        }
    }
```

Reply:
음 비교값이라는 의미로 사용했는데 더 적절한 표현이 있을까요? 일단 judgeSelectCellEnabled 함수는 사라져서 judgeBottomButtonEnabled에서 활용될 것 같습니다. targetValue? .. targetSelectedCellCount..?

Reply:
적절해보이네요 ~~~

Reply:
어떤게 적절하다는 건가용 targetValue, targetaSelectedCellCount, comparisonValue..?

Reply:
[eb90478](https://github.com/DeveloperAcademy-POSTECH/MacC-Team-Vegeting/pull/143/commits/eb90478b7cb92ce07cc6dcd8fb94d1f8398bc2d0) targetValue로 반영하였습니다.